types/jest/index.d.ts

```diff
@@ -667,25 +677,40 @@ declare namespace jest {
     interface Matchers<R, T> {
         /**
          * Ensures the last call to a mock function was provided specific args.
+         *
+         * Optionally, you can provide a type for the expected arguments via a generic.
+         * Note that the type must be either an array or a tuple.
          */
-        lastCalledWith(...args: any[]): R;
+        lastCalledWith<E extends any[]>(...args: E): R;
         /**
          * Ensure that the last call to a mock function has returned a specified value.
+         *
+         * Optionally, you can provide a type for the expected value via a generic.
+         * This is particuarly useful for ensuring expected objects have the right structure.
          */
-        lastReturnedWith(value: any): R;
+        lastReturnedWith<E>(value: E): R;
```

Comment:
```suggestion
        lastReturnedWith<E = any>(value: E): R;
```

Explicit fallback rather than implicit